pgmpy/models/MarkovModel.py

```diff
@@ -190,14 +198,22 @@ def get_cardinality(self):
         >>> factor = DiscreteFactor(['Alice', 'Bob'], cardinality=[2, 2],
         ...                 values=np.random.rand(4))
         >>> student.add_factors(factor)
+        >>> student.get_cardinality(node='Alice')
+        2
         >>> student.get_cardinality()
         defaultdict(<class 'int'>, {'Bob': 2, 'Alice': 2})
         """
-        cardinalities = defaultdict(int)
-        for factor in self.factors:
-            for variable, cardinality in zip(factor.scope(), factor.cardinality):
-                cardinalities[variable] = cardinality
-        return cardinalities
+        if node:
+            for factor in self.factors:
+                for variable, cardinality in zip(factor.scope(), factor.cardinality):
+                    if node == variable:
+                       return cardinality
+        else:
+            cardinalities = defaultdict(int)
```

Comment:
@lohani2280 Is there any particular need for defaultdict?

Reply:
@khalibartan This method was implemented using `defaultdict` only ,so i didn't  thought of changing it. Now, after doing some experiments, i think a normal dict would also suffice this implementation.
Its because, `defaultdict` is similar to the usual dictionary (dict) container, but it has one difference:  that it is initialized with a function  that takes no arguments and provides the default value for a nonexistent key. Thus , a defaultdict will never raise a `KeyError` if you try to get an item with a key that is not currently in the dictionary unlike usual dict container.
Now ,here in this method we are always appending `variables` and its corresponding `cardinality` into the dict container.We are never retrieving a value using a key(either an existing or non-existing key). So ,if their is no retrieval then a normal dict would itself suffice . Please correct me if i am wrong.

Reply:
@khalibartan @ankurankan I think one more thing is lacking in this implementation . If `node` passed while calling get_cardinality is not present in the model then it should throw an error. Right now this is not happening.

Reply:
@lohani2280 I agree with you. Could you please create an issue for this ?

Reply:
@ankurankan Do you want me to create an issue regarding `defaultdict` implementation or regarding ` If node passed while calling get_cardinality, is not present in the model then it should throw an error. Right now this is not happening.`  ?

Reply:
@lohani2280 Both.